### DISCUSSION TITLE
video14

### DISCUSSION THREAD

**Discussion by mcharlie9** [jacekschae/learn-pedestal-course-files#10] [Feedback]
At 0:15, you mentioned that when going to localhost:3000, everything will be working.

However, after starting the server, the routes don't seem to be working. For instance, GET /recipes returns 404 "Not Found", instead of "list recipes". Would you mind checking if you are also seeing the same thing?  Thanks.

**Comment by jacekschae** [comment]
Yeah. Planning to dig into this in the next video. I think the main problem is the config in the router. If you update as following, it should work. Note the `:host`.

```clojure

(def routes
  (route/expand-routes
    #{{:app-name :cheffy :schema :http :host "localhost"}
      ["/recipes" :get list-recipes :route-name :list-recipes]
      ["/recipes" :post upsert-recipe :route-name :create-recipe]
      ["/recipes/:recipe-id" :put upsert-recipe :route-name :update-recipe]}))

```

**Reply by mcharlie9** [reply]
Thanks for the hint!  Yes, the routes work after updating the :host value to "localhost".

It would be a valuable addition to the course if, in the next video, there can be an in-depth explanation regarding the service map which is an important but rather abstract idea of Pedestal.

In the official docs, it is mentioned that "the service map provides a configuration that Pedestal turns into a service function, server function, chain provider, router, routes, and default set of interceptors.".  In other words, the service map is used for quite a few things. For instance, I am not sure which of theses things is/are related to the :host key.  I can see that the :host value is used to configure the routes, since without setting "localhost" as :host, the routes do not work.  However, even with an invalid :host value, the server is able to be started without issue "automatically" in localhost, despite the routes not working.  Seems like the :host key has no effect on the server function; or maybe, an invalid :host value will be replaced by the default "localhost" when starting the server?

This is a question mark that popped up while I was working though the videos.  It would be very nice if the course can provide insights into such questions that are not easily solvable by googling and may require tacit knowledge and digging into the codes.  I understand it is difficult to give the students all the fish (i.e. explanations), so instead, any sharing that teaches the students by example how to fish (i.e. suggestions on how best to understand Pedestal on one's own) would make the course uniquely valuable.

Anyway, the above is just a suggestion for your consideration. Again, great job with the videos, can't wait to go through another new set of videos, thanks!

P.S. Also found a typo in the codes: should be :scheme instead of :schema.

**Reply by jacekschae** [reply]
Thanks for your comment. I'll take this into consideration when recording the next set of videos.

**Reply by mcharlie9** [reply]
Just looking at the codes, and noticed the (def routes ...) is still showing :schema instead of the correct :scheme.  Can cross the bridge when we come to it, but I do wonder when this keyword would be used and whether it would have any "side effect" as the course continues.

**Reply by jacekschae** [reply]
I should have fixed it within 21. Shame that I missed this one. I'll circle back to this one.

**Reply by mcharlie9** [reply]
Just spent quite some time on video21 since I cannot get the pedestal test to return 200.  Finally, I realize that by changing :schema to the correct :scheme, the pedestal test would return 500 instead. Seems like we have already come to the bridge regarding this typo. It is interesting that the codes with the typo work because the typo KV pair is ignored and the default value is used. I wonder why the code is acting up... 

Steps to reproduce: 
1) Change :schema to :scheme 
2) (restart-dev) 
3) (pt/response-for ...) 

Would you mind taking a look to see if you are also getting 500 error with the above?  It would be nice to figure out why this is the case.

**Reply by jacekschae** [reply]
Same things. If you fully namespace `:scheme` to `::http/scheme` the int's all fine. I'll probably take a second take on the 21.

**Reply by mcharlie9** [reply]
Just figured it out too. Thanks for letting me know.  Yes, a second take might be simpler. Will send through a few more feedbacks regarding video21 on a new discussion thread.
